### PR TITLE
Codesniffer: setcookie sniff

### PR DESCRIPTION
**Pull request by kraftbj**
- Add new rule for setcookie
- PHPCS for SSO

Addresses p3btAN-1K8-p2

#### Changes proposed in this Pull Request:

* Adds a new sniff for httponly.

#### Jetpack product discussion

p3btAN-1K8-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* In addition to the tests for the codesniffer itself, run through SSO and check that httponly is set for the cookies.

### CODE REVIEW

File: projects/packages/codesniffer/Jetpack/Sniffs/Functions/SetCookieSniff.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<?php
```
Comment (bindlegirl) [inline projects/packages/codesniffer/Jetpack/Sniffs/Functions/SetCookieSniff.php:1]:
```suggestion
<?php //phpcs:ignore WordPress.Files.FileName.NotHyphenatedLowercase
```
PHPCS in VScode complained about this.

**Comment by bindlegirl** [comment]
SSO & httponly work well for me.

**Review by jeherve — APPROVED** [review]
This tests well for me, it should be good to merge!

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-10.9` in 335e8074b44209bf2124c65c7c80d6ea94b1451f

**Comment by samiff** [comment]
@kraftbj Is any action needed with D81047-code or can that be abandoned?

**Comment by kraftbj** [comment]
r247545-wpcom

**Pull request merged**